lib/pry/config/behavior.rb

```diff
@@ -188,6 +188,8 @@ def method_missing(name, *args, &block)
   end
 
   def respond_to_missing?(key, include_all=false)
+    key = key.to_s
+    key = key[0..-2] if key.end_with? ASSIGNMENT
```

Comment:
i dont know much about the config system anymore, but why doesn't `@default` just respond to `foo=` ? why do we have to clip off the final `=` ?

Reply:
All instances of `Pry::Config` are a hash table implemented on top of method_missing 
 - although the methods are defined lazily on first access. `foo=` doesn't exist until then, so that's what respond_to_missing? is accounting for, before first access.

Reply:
> why do we have to clip off the final = ?

because the hash table doesn't store the '=' character, but it's safe to assume they come in pairs.

Reply:
Pry::Config::Default is a more specialised case, the last in the chain, that defines regular methods but `Pry.config.exception_whitelist=` delegates to it. `Pry.config` is a `Pry::Config`.
`_pry_.config => Pry.config => Pry::Config::Default`. something like that.